### DISCUSSION TITLE
How to have dev-only routes?

### DISCUSSION THREAD

**Discussion by gvanto** [pimcore/pimcore#13936] [Q&A / Support]
I would like to have a TestController which only loads in the dev environment.

I know routes can loaded via yaml files, but how to load conditionally in dev env only?
(or have a class load the routes if possible and [autoload-dev this class](https://github.com/pimcore/pimcore/discussions/13918) in composer?)

**Comment by jdreesen** [comment]
Since Pimcore builds upon Symfony, there are several possibilities:

- define the route in a YAML that's only loaded in `dev` env, e.g. [`config/routes/dev/my-routes.yaml`](https://github.com/pimcore/pimcore/blob/10.5/lib/Kernel.php#L149)
- define it under the [special `when@dev:` key](https://symfony.com/blog/new-in-symfony-5-3-configure-multiple-environments-in-a-single-file) in your `config/routes.yaml`
- define it with a [`condition: "%kernel.environment% === 'dev'"`](https://symfony.com/doc/current/routing.html#matching-expressions)

and possibly even more.

**Reply by gvanto** [reply]
Thanks @jdreesen I have tried using routing_dev.yml (using Symfony 3.4 and pimcore 5.8) as per [SO post](https://stackoverflow.com/questions/28253545/symfony-defining-a-route-only-in-dev-environment )

Strangely enough adding my test route to Resources/config/routing_dev.yml registers the route (shown in
 routes list with ./bin/console debug:router) but the route doesn't work (get a 404 error). 

Using the 'condition' for now but ideally would like to use routing_dev.yml as it's considered best practice.
